doc/page-cmakefunctions.h

```diff
@@ -0,0 +1,173 @@
+
+\param PLUGIN_NAME name of a plugin
+\param URI uri of qml plugin
+\param VERSION qml plugin version , if not given set it to 1.0
+\param OUTPUT_BASE_DIRECTORY subpath in CMAKE_BINARY_DIR, where artifacts ( libraries, binary, qmldir etc)  are going to get installed. If not given default is "imports"
+\param SOURCES list of cpp files for the plugin
+\param HEADERS list of h files for the plugin
+\param SOURCES_GLOB_RECURSE list of cpp files for the plugin, could be provided as a wildcard expression
+\param HEADERS_GLOB_RECURSE list of h files for the plugin, could be provided as a wildcard expression
+\param LINK_LIBRARIES targets to be linked with
+
+<hr>
+
+
+\code
+function(facelift_add_interface TARGET_NAME)
+\endcode
+
+Registers a new facelift interface and generates corresponding modules: ipc, types, devtools, etc.
+
+\param TARGET_NAME name of the facelift interface
+\param INTERFACE_DEFINITION_FOLDER folder with qface definition files
+\param LINK_LIBRARIES targets to be linked with
+\param IMPORT_FOLDERS import interfaces from the folder paths
+
+<hr>
+
+
+\code
+function(facelift_generateQRC OUTPUT_FILE)
+\endcode
+Generates the qrc resource file from the given input folders. Function will look for js, qml, ttf , png, sci, json, astcz, tcsh, tesh, gsh, fsh files.
+\param OUTPUT_FILE  output qrc resource file
+\param INPUT_FOLDERS list of folders containing resources to be parsed
+
+<hr>
+
+\code
+function(facelift_add_library TARGET_NAME)
+\endcode
+
+\param TARGET_NAME name of the library
+\param LIBRARY_TYPE type of library shared static
+\param SOURCES sources files compiled to create a library
+\param HEADERS header files compiled to create a library
+\param LINK_LIBRARIES  external library names to be included
+\param SOURCES_GLOB_RECURSE list of cpp files for the plugin, could be provided as a wildcard expression
+\param HEADERS_GLOB_RECURSE list of h files for the plugin, could be provided as a wildcard expression
+\param EXPORT Create the directory for the installed headers
+\param INSTALL Install library
+
+<hr>
+
+
+\code
+function(facelift_add_executable TARGET_NAME)
+\endcode
+Add an executable to the project using the specified source files.
+\param TARGET_NAME name of the executable file
+
+<hr>
+
+\code
+function(facelift_add_test TEST_NAME)
+\endcode
+Adds the test with TEST_NAME as test case name
+\param TEST_NAME name of the test case
+\param SOURCES source cpp file 
+\param HEADERS header file
+\param LINK_LIBRARIES link library if needed
+
+<hr>
+
+
+
+\section private-functions Functions for FaceLift maintainers
+
+These functions are meant to be used by FaceLift maintainers, a.k.a by FaceLift internally.
+
+\code
+function(facelift_add_unity_files TARGET_NAME VAR_NAME)
+\endcode
+Generates unity files for targetname provided in the function
+\param TARGET_NAME target name of the file
+\param VAR_NAME  optional
+<hr>
+
+\code
+function(facelift_synchronize_folders FOLDER_SOURCE FOLDER_DESTINATION)
```

Comment:
not public

Reply:
its not public , i am putting these in private section